src/BiDirectionalLinkedListImp.h

```diff
@@ -0,0 +1,152 @@
+//
+// Created by shvmo on 02/05/2020.
+//
+
+#ifndef DATASTRUCTURESWET1_BIDIRECTIONALLINKEDLISTIMP_H
+#define DATASTRUCTURESWET1_BIDIRECTIONALLINKEDLISTIMP_H
+
+#include <cassert>
+#include "BiDirectionalLinkedList.h"
+
+using std::make_shared;
+
+template<class Type>
+BiDirectionalLinkedList<Type>::BiDirectionalLinkedList() = default;
+
+template<class Type>
+BiDirectionalLinkedList<Type>::~BiDirectionalLinkedList() = default;
+
+template<class Type>
+shared_ptr<BiDirectionalNode<Type>> BiDirectionalLinkedList<Type>::getFirst() const {
+    return this->first;
+}
+
+template<class Type>
+BiDirectionalNode<Type> *BiDirectionalLinkedList<Type>::getFirstPointer() const {
+    return this->first.get();
+}
+
+template<class Type>
+shared_ptr<BiDirectionalNode<Type>> BiDirectionalLinkedList<Type>::getLast() const {
+    return this->last;
+}
+
+template<class Type>
+void BiDirectionalLinkedList<Type>::insert(const Type &newValue) {
+    if (this->first == nullptr) {
+        this->first = make_shared<BiDirectionalNode<Type>>();
+
+        this->first->setNext(nullptr);
+        this->first->setPrevious(nullptr);
+        this->first->setValue(newValue);
+
+        this->last = this->first;
+    } else {
+        shared_ptr<BiDirectionalNode<Type>> newLast = make_shared<BiDirectionalNode<Type>>();
+
+        newLast->setNext(nullptr);
+        newLast->setPrevious(last);
```

Comment:
`newLast->setPrevious(last);` 
to 
`newLast->setPrevious(this->last);`
please

Reply:
or, alternatively, we can remove the `this->`'s from the `if` section.